### PR TITLE
Add domain & codomain to transform

### PR DESCRIPTION
**Pull request by tczorro**
1. Add domain and codomain property to each transform
2. Change tests accordingly

### CODE REVIEW

File: src/grid/rtransform.py [hunk 1]
```diff
@@ -52,7 +52,17 @@ def deriv2(self, x):
...
-    def transform_1d_grid(self, oned_grid):
+    @property
+    def tf_domain(self):
+        """tuple: Transformation domain."""
+        return self._tf_domain
+
+    @property
+    def tf_codomain(self):
+        """tuple: Transformation codomain."""
+        return self._tf_codomain
```
Comment (FarnazH) [inline src/grid/rtransform.py:63]: Isn't it simpler ans easier to remember to just call them `domain` and `codomain`?

File: src/grid/rtransform.py [hunk 2]
```diff
@@ -77,6 +87,15 @@ def transform_1d_grid(self, oned_grid):
         """
         if not isinstance(oned_grid, OneDGrid):
             raise TypeError(f"Input grid is not OneDGrid, got {type(oned_grid)}")
+        # check domain
+        if (
+            oned_grid.domain[0] < self.tf_domain[0]
+            or oned_grid.domain[1] > self.tf_domain[1]
+        ):
+            raise ValueError(
+                "Input grid domain do not match with tf domain.\n"
```
Comment (FarnazH) [inline src/grid/rtransform.py:96]: To be more clear for the user, we can say:
"Given 1D grid domain does not match the transformation domain"

**Comment by FarnazH** [comment]
Thanks @tczorro. Please see my minor comments above. This PR address issue #101 by raising an informative error when transforming the 1D grid.

File: src/grid/rtransform.py (hunk 2 above)
Comment (tczorro) [reply src/grid/rtransform.py:96]: I also prefer `domain` and `codomain`. Just not sure would it bring ambiguity with the `domain` in grid.

**Comment by tczorro** [comment]
Just to double check, if no further question, i will merge this PR tomorrow (Aug 12th) in case the changes become obsolete.

**Comment by FarnazH** [comment]
Thanks for the update @tczorro. I am have no further comments.

**Pull request merged**